visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -233,6 +246,54 @@ export default {
         },
         formatDate(date, format=this.dateFormat) {
             return window.Moment(date).format(format);
+        },
+        handleDefaults() {
+            const urlPreset = this.getFilterValue("date_preset");
+
+            // Handle Preset, override dates.
+            if (!urlPreset && this.defaultDatePreset) {
+                // If preset used, skip start and end dates default
+                return this.handleDefaultPreset();
+            }
+
+            return this.handleDefaultDates();
+        },
+        handleDefaultPreset() {
+            const defaultPreset = this.presets[this.mode]
+                .filter(p => {
+                    return p.key === this.defaultDatePreset ||
+                        p.label === this.defaultDatePreset
+                })[0];
+            
+            if (defaultPreset) {
+                this.handlePreset(defaultPreset);
+                this.handleConfirm(this.date);
+            }
+        },
+        handleDefaultDates() {
+            const urlStartDate = this.getFilterValue("start_date");
+            const urlEndDate = this.getFilterValue("end_date");
+
+            if (urlStartDate || urlEndDate) {
+                return;
+            }
+
+            const defaultStartDate = this.defaultStartDate;
+            const defaultEndDate = this.defaultEndDate || defaultStartDate; // If no end date default, set same as start to avoid error.
```

Comment:
If 'defaultStartDate' is not supplied it will be defaulted to an empty (via the props 'default'). This means the `defaultEndDate` will also be an empty string. Is that ok?

Reply:
Yes, because if both are empty, the whole default logic will be skipped.